cmake/ElektraCompiling.cmake

```diff
@@ -142,6 +142,7 @@ set (COMMON_FLAGS "${COMMON_FLAGS} -Wformat -Wformat-security")
 set (COMMON_FLAGS "${COMMON_FLAGS} -Wshadow")
 set (COMMON_FLAGS "${COMMON_FLAGS} -Wcomments -Wtrigraphs -Wundef")
 set (COMMON_FLAGS "${COMMON_FLAGS} -Wuninitialized -Winit-self")
+set (COMMON_FLAGS "${COMMON_FLAGS} -Werror")
```

Comment:
We cannot add `-Werror` here. For production builds it is always useful to *not* have `-Werror`. We can never know if a later version of a compiler will throw additional warnings. We do not want to fail in such situations.